qcelemental/molparse/from_schema.py

```diff
@@ -143,7 +154,7 @@ def contiguize_from_fragment_pattern(frag_pattern, *, geom=None, verbose:int=1,
             """Error: QCElemental would need to reorder atoms to accommodate non-contiguous fragments""")
 
     if geom is not None:
-        ncgeom = np.array(geom).reshape(-1, 3)
+        ncgeom = np.asarray(geom).reshape(-1, 3)
```

Comment:
I suppose I expect a constructor to return an independent-memory object. Can't cite any particular danger, though. (note to self: l157 and l138)

Reply:
See below, we can figure that one out.

Reply:
sounds good on below. I like having a `copy=True/False` on all to/from rather than trying to foresee usage patterns. I was wishing for `const` for a bit. Freely admit that I err on the don't-surprise-me side of mem alloc, but now seems a good time to tighten up.

Reply:
After hunting through this we added copy pretty much everywhere it is useful. I think I found the last ones... let me know.

Going to let the current `__init__` copy once.

Reply:
My last reservation is that as-is `from_schema` returns mem-independent `geom` if `nfr>1` and mem-dep `geom` if `nfr==1`. If you're ok with that, I'm good with the PR.

Reply:
We have:
```
    if geom is not None:
        ncgeom = np.array(geom).reshape(-1, 3)
```

I don't see the pathway? Happy to fix.

Reply:
two issues:

* nfr=1 returns the raw argument https://github.com/MolSSI/QCElemental/pull/86/files#diff-2d5d131c2b409ebc837aa226ab68e39fR138
whereas https://github.com/MolSSI/QCElemental/pull/86/files#diff-2d5d131c2b409ebc837aa226ab68e39fL149 is surely copying. unless i'm reading it wrong ...

* I think the new block https://github.com/MolSSI/QCElemental/pull/86/files#diff-2d5d131c2b409ebc837aa226ab68e39fR134-R143 needs to move down below the three checks https://github.com/MolSSI/QCElemental/pull/86/files#diff-2d5d131c2b409ebc837aa226ab68e39fR155 because that middle one (despite the frag-focussed message) also catches single-frag [1, 3, 2] cases.

Reply:
Rewritten slightly to make the first path a shortcut and I believe properly checks for it.